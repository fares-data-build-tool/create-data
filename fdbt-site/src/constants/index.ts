export const OPERATOR_COOKIE = 'fdbt-operator';

export const FARE_TYPE_COOKIE = 'fdbt-fare-type';

export const PASSENGER_TYPE_COOKIE = 'fdbt-passenger-type';

export const SERVICE_COOKIE = 'fdbt-service';

export const JOURNEY_COOKIE = 'fdbt-journey';

export const GOVUK_LINK = 'https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/';

export const FARE_STAGES_COOKIE = 'fdbt-fare-stages';

export const STAGE_NAME_VALIDATION_COOKIE = 'fdbt-stage-names-validation';

export const FEEDBACK_LINK = 'mailto:fdbt@transportforthenorth.com?bcc=tfn-feedback@infinityworks.com';

export const STAGE_NAMES_COOKIE = 'fdbt-stage-names';

export const DAYS_VALID_COOKIE = 'fdbt-days-valid';

export const PRODUCT_DETAILS_COOKIE = 'fdbt-product-details';

export const PERIOD_TYPE_COOKIE = 'fdbt-period-type';

export const CSV_UPLOAD_COOKIE = 'fdbt-csv-upload';

export const CSV_ZONE_UPLOAD_COOKIE = 'fdbt-csv-zone-upload';

export const PERIOD_EXPIRY_COOKIE = 'fdbt-period-expiry';

export const SERVICE_LIST_COOKIE = 'fdbt-services';

export const INPUT_METHOD_COOKIE = 'fdbt-input-method';

export const NUMBER_OF_STAGES_COOKIE = 'fdbt-number-stages';

export const MATCHING_COOKIE = 'fdbt-matching';

export const MULTIPLE_SERVICE_COOKIE = 'fdbt-multiple-service';

export const NUMBER_OF_PRODUCTS_COOKIE = 'fdbt-number-of-products';

export const MULTIPLE_PRODUCT_COOKIE = 'fdbt-multiple-product';

export const ALL_COOKIES: string[] = [
    'fdbt-operator',
    'fdbt-fare-type',
    'fdbt-passenger-type',
    'fdbt-service',
    'fdbt-journey',
    'fdbt-fare-stages',
    'fdbt-stage-names-validation',
    'fdbt-stage-names',
    'fdbt-price-entry',
    'fdbt-days-valid',
    'fdbt-product-details',
    'fdbt-period-type',
    'fdbt-csv-upload',
    'fdbt-csv-zone-upload',
    'fdbt-number-of-products',
    'fdbt-period-expiry',
    'fdbt-services',
    'fdbt-input-method',
    'fdbt-matching',
    'fdbt-number-stages',
    'fdbt-multiple-service',
    'fdbt-number-of-products',
    'fdbt-multiple-product',
];

export const ALLOWED_CSV_FILE_TYPES = [
    'text/plain',
    'text/x-csv',
    'application/vnd.ms-excel',
    'application/csv',
    'application/x-csv',
    'application/octet-stream',
    'text/csv',
    'text/comma-separated-values',
    'text/x-comma-separated-values',
    'text/tab-separated-values',
];

export const STAGE = process.env.STAGE || 'dev';

export const RAW_USER_DATA_BUCKET_NAME = `fdbt-raw-user-data-${STAGE}`;
export const USER_DATA_BUCKET_NAME = `fdbt-user-data-${STAGE}`;
export const MATCHING_DATA_BUCKET_NAME = `fdbt-matching-data-${STAGE}`;
