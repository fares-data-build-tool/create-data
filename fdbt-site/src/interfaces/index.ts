export interface PeriodProductType {
    productName: string;
    productPrice: string;
    productNameError: boolean;
    productPriceError: boolean;
}
