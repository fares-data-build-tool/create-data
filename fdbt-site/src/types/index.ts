export interface ErrorInfo {
    errorMessage: string;
    id: string;
}
