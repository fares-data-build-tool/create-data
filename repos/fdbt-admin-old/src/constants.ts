export const ATTRIBUTE_MAP: { [key: string]: string } = {
    'custom:noc': 'NOC',
    'custom:schemeOperator': 'Scheme Name',
    'custom:schemeRegionCode': 'Scheme Region',
};

export const STATUS_MAP: { [key: string]: string } = {
    CONFIRMED: 'Registered',
    FORCE_CHANGE_PASSWORD: 'Awaiting Registration',
};

export const MAIN_USER_POOL_PREFIX = 'fdbt-user-pool';
export const NETEX_DATA_BUCKET_PREFIX = 'fdbt-netex-data';
export const PRODUCTS_DATA_BUCKET_PREFIX = 'fdbt-products-data';

export const AWS_REGION = 'eu-west-2';
