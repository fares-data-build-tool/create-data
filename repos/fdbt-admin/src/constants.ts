export const ATTRIBUTE_MAP: { [key: string]: string } = {
    'custom:noc': 'NOC',
    'custom:schemeOperator': 'Scheme Name',
    'custom:schemeRegionCode': 'Scheme Region',
};

export const STATUS_MAP: { [key: string]: string } = {
    CONFIRMED: 'Registered',
    FORCE_CHANGE_PASSWORD: 'Awaiting Registration',
};

export const MAIN_USER_POOL_PREFIX = 'fdbt-user-pool';

export const AWS_REGION = 'eu-west-2';
