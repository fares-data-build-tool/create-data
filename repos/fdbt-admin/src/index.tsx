import { StrictMode } from 'react';
import ReactDOM from 'react-dom';
import Amplify from 'aws-amplify';
import App from './App';
import awsconfig from './aws-exports';

import './index.css';

Amplify.configure(awsconfig);

ReactDOM.render(
    <StrictMode>
        <App />
    </StrictMode>,
    document.getElementById('root'),
);
