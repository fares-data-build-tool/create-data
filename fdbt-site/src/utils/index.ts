import Cookies from 'cookies';
import { NextPageContext } from 'next';
import { IncomingMessage } from 'http';
import axios from 'axios';
import { parseCookies, destroyCookie } from 'nookies';
import { decode } from 'jsonwebtoken';
import { OPERATOR_COOKIE, ID_TOKEN_COOKIE, REFRESH_TOKEN_COOKIE } from '../constants/index';
import { Stop } from '../data/auroradb';
import { ErrorInfo, CognitoIdToken } from '../interfaces';

export const setCookieOnServerSide = (ctx: NextPageContext, cookieName: string, cookieValue: string): void => {
    if (ctx.req && ctx.res) {
        const cookies = new Cookies(ctx.req, ctx.res);
        const host = ctx?.req?.headers?.host;
        const domain = host ? host.split(':')[0] : '';

        cookies.set(cookieName, cookieValue, { domain, path: '/' });
    }
};

export const deleteCookieOnServerSide = (ctx: NextPageContext, cookieName: string): void => {
    if (ctx.req && ctx.res) {
        const cookies = new Cookies(ctx.req, ctx.res);
        const host = ctx?.req?.headers?.host;
        const domain = host ? host.split(':')[0] : '';

        cookies.set(cookieName, '', { overwrite: true, maxAge: 0, domain, path: '/' });
    }
};

export const deleteAllCookiesOnServerSide = (ctx: NextPageContext): void => {
    const cookies = parseCookies(ctx);
    const cookieWhitelist = [OPERATOR_COOKIE, ID_TOKEN_COOKIE, REFRESH_TOKEN_COOKIE];

    Object.keys(cookies).forEach(cookie => {
        if (!cookieWhitelist.includes(cookie)) {
            destroyCookie(ctx, cookie);
        }
    });
};

export const getHost = (req: IncomingMessage | undefined): string => {
    if (!req) {
        return '';
    }
    const host = req?.headers?.host;

    if (host) {
        if (host && host.startsWith('localhost')) {
            return `http://${host}`;
        }
        return `https://${host}`;
    }

    return '';
};

export const isSessionValid = async (url: string, req: IncomingMessage | undefined): Promise<boolean> => {
    try {
        const response = await axios.get(url, {
            withCredentials: true,
            headers: {
                Cookie: req?.headers.cookie,
            },
        });

        return response.data.Valid;
    } catch (error) {
        return false;
    }
};

export const getUuidFromCookies = (ctx: NextPageContext): string | null => {
    const cookies = parseCookies(ctx);
    const operatorCookie = cookies[OPERATOR_COOKIE];
    if (!operatorCookie) {
        return null;
    }
    const operatorInfo = JSON.parse(operatorCookie);
    return operatorInfo.uuid;
};

export const getJourneyPatternFromCookies = (ctx: NextPageContext): string | null => {
    const cookies = parseCookies(ctx);
    const operatorCookie = cookies[OPERATOR_COOKIE];
    if (!operatorCookie) {
        return null;
    }
    const operatorInfo = JSON.parse(operatorCookie);
    return operatorInfo.journeyPattern;
};

export const formatStopName = (stop: Stop): string =>
    `${stop.localityName ? `${stop.localityName}, ` : ''}${stop.indicator ?? ''} ${stop.stopName ?? ''}${
        stop.street ? ` (on ${stop.street})` : ''
    }`;

export const buildTitle = (errors: ErrorInfo[], title: string): string => {
    if (errors.length > 0) {
        return `Error: ${title}`;
    }

    return title;
};

export const getAttributeFromIdToken = <T extends keyof CognitoIdToken>(
    ctx: NextPageContext,
    attribute: T,
): CognitoIdToken[T] | null => {
    const cookies = parseCookies(ctx);
    const idToken = cookies[ID_TOKEN_COOKIE];

    if (!idToken) {
        return null;
    }

    const decodedIdToken = decode(idToken) as CognitoIdToken;

    return decodedIdToken[attribute] ?? null;
};

export const getNocFromIdToken = (ctx: NextPageContext): string | null => getAttributeFromIdToken(ctx, 'custom:noc');
