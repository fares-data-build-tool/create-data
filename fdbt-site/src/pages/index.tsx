import React, { ReactElement } from 'react';
import TwoThirdsLayout from '../layout/Layout';
import { CustomAppProps, NextPageContextWithSession } from '../interfaces';
import { checkIfMultipleOperators } from '../utils';

const title = 'Create Fares Data';
const description = 'Create Fares Data is a service that allows you to generate data in NeTEx format';

interface StartProps {
    multipleOperators: boolean;
}

const Start = ({ multipleOperators }: StartProps & CustomAppProps): ReactElement => (
    <TwoThirdsLayout title={title} description={description}>
        <h1 className="govuk-heading-xl">Create Fares Data</h1>

        <p className="govuk-body">
            This service is for creating or accessing NeTEx data for public transport services, excluding rail, in
            England. The service can be used by:
        </p>
        <ul className="govuk-list govuk-list--bullet">
            <li>Bus operators running commercial services in England</li>
            <li>Local transport authorities acting on behalf of bus operators</li>
            <li>Local transport authorities that operate their own services</li>
        </ul>

        <p className="govuk-body">Use this service to:</p>
        <ul className="govuk-list govuk-list--bullet">
            <li>Generate fares data in NeTEx format for a new or existing service</li>
            <li>Download your own fares data</li>
        </ul>
        <p className="govuk-body">
            All service data is taken directly from the Traveline National Data Set and the NaPTAN database. To avoid
            data discrepancies, ensure that all service data is correct within these datasets before using the service.
        </p>

        <a
            href={multipleOperators ? '/multipleOperators' : '/fareType'}
            role="button"
            draggable="false"
            className="govuk-button govuk-button--start"
            data-module="govuk-button"
            id="start-now-button"
        >
            Start now
            <svg
                className="govuk-button__start-icon"
                xmlns="http://www.w3.org/2000/svg"
                width="17.5"
                height="19"
                viewBox="0 0 33 40"
                role="presentation"
                focusable="false"
            >
                <path fill="currentColor" d="M0 0h13l20 20-20 20H0l20-20z" />
            </svg>
        </a>
    </TwoThirdsLayout>
);

export const getServerSideProps = (ctx: NextPageContextWithSession): { props: StartProps } => ({
    props: { multipleOperators: checkIfMultipleOperators(ctx) },
});

export default Start;
