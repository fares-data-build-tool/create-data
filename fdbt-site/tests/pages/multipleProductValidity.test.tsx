import { shallow } from 'enzyme';
import React from 'react';
import MultiProductValidity, { getServerSideProps } from '../../src/pages/multipleProductValidity';
import { getMockContext } from '../testData/mockData';
import { NUMBER_OF_PRODUCTS_ATTRIBUTE, MULTIPLE_PRODUCT_ATTRIBUTE } from '../../src/constants';

describe('pages', () => {
    describe('multipleProductValidity', () => {
        const multipleProducts = [
            {
                productName: 'Weekly Ticket',
                productNameId: 'multiple-product-name-1',
                productPrice: '50',
                productPriceId: 'multiple-product-price-1',
                productDuration: '5',
                productDurationId: 'multiple-product-duration-1',
            },
            {
                productName: 'Day Ticket',
                productNameId: 'multiple-product-name-2',
                productPrice: '2.50',
                productPriceId: 'multiple-product-price-2',
                productDuration: '1',
                productDurationId: 'multiple-product-duration-2',
            },
        ];

        it('should render correctly', () => {
            const wrapper = shallow(
                <MultiProductValidity
                    operator="Infinity Line"
                    passengerType="Adult"
                    numberOfProducts="2"
                    multipleProducts={multipleProducts}
                    errors={[]}
                    csrfToken=""
                    pageProps={[]}
                />,
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('should render error messaging when errors are passed', () => {
            const wrapper = shallow(
                <MultiProductValidity
                    operator="Infinity Line"
                    passengerType="Adult"
                    numberOfProducts="2"
                    multipleProducts={multipleProducts}
                    errors={[
                        {
                            errorMessage: 'Select one of the two validity options',
                            id: 'multiple-product-validity-error',
                        },
                    ]}
                    csrfToken=""
                    pageProps={[]}
                />,
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('renders 2 radio buttons per product', () => {
            const wrapper = shallow(
                <MultiProductValidity
                    operator="Infinity Line"
                    passengerType="Adult"
                    numberOfProducts="2"
                    multipleProducts={multipleProducts}
                    errors={[]}
                    csrfToken=""
                    pageProps={[]}
                />,
            );
            expect(wrapper.find('.govuk-radios__item')).toHaveLength(4);
        });

        describe('getServerSideProps', () => {
            it('should return number of products to display, name of operator and products if they are set in the cookie', () => {
                const ctx = getMockContext({
                    session: {
                        [NUMBER_OF_PRODUCTS_ATTRIBUTE]: { numberOfProductsInput: '2' },
                    },
                });
                const result = getServerSideProps(ctx);

                expect(result.props.numberOfProducts).toBe('2');
                expect(result.props.operator).toBe('test');
                expect(result.props.multipleProducts[0].productName).toBe('Weekly Ticket');
                expect(result.props.multipleProducts[0].productPrice).toBe('50');
                expect(result.props.multipleProducts[0].productDuration).toBe('5');
                expect(result.props.multipleProducts.length).toBe(3);
            });

            it('should throw an error if the necessary cookies to render the page are not present', () => {
                const ctx = getMockContext({
                    cookies: {
                        operator: null,
                        multipleProduct: null,
                    },
                });
                expect(() => getServerSideProps(ctx)).toThrow(
                    'Necessary cookies/session not found to display the multiple product validity page',
                );
            });

            it('returns errors in the props if there are validity errors on the product object', () => {
                const ctx = getMockContext({
                    session: {
                        [MULTIPLE_PRODUCT_ATTRIBUTE]: {
                            products: [
                                {
                                    productName: 'Best Product',
                                    productNameId: 'multipleProductNameInput0',
                                    productPrice: '2',
                                    productPriceId: 'multipleProductPriceInput0',
                                    productDuration: '3',
                                    productDurationId: 'multipleProductDurationInput0',
                                    productValidity: '',
                                    productValidityError: 'Select one of the two validity options',
                                },
                                {
                                    productName: 'Super Product',
                                    productPrice: '3',
                                    productDuration: '4',
                                    productValidity: '24hr',
                                },
                            ],
                        },
                        [NUMBER_OF_PRODUCTS_ATTRIBUTE]: { numberOfProductsInput: '2' },
                    },
                });

                const result = getServerSideProps(ctx);

                expect(result.props.multipleProducts.length).toBe(2);
                expect(result.props.errors.length).toBe(1);
                expect(result.props.errors[0].errorMessage).toBe('Select one of the two validity options');
                expect(result.props.errors[0].id).toBe('multiple-product-validity-radios-error');
            });
        });
    });
});
