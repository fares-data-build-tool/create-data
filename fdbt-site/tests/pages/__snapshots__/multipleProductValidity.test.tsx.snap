// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages multipleProductValidity should render correctly 1`] = `
<Layout
  description="Multiple Product Validity selection page of the Fares Data Build Tool"
  title="Multiple Product Validity - Fares Data Build Tool"
>
  <main
    className="govuk-main-wrapper app-main-class multiple-product-validity-page"
    id="main-content"
    role="main"
  >
    <form
      action="/api/multipleProductValidity"
      method="post"
    >
      <ErrorSummary
        errors={Array []}
      />
      <div
        className="govuk-form-group "
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--xl"
        >
          <h1
            className="govuk-fieldset__heading"
            id="multiple-product-validity-page-heading"
          >
            When does the product expire?
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="operator-products-hint"
        >
          Infinity Line
           - 
          2
           products - 
          Adult
        </span>
        <span
          className="govuk-hint"
          id="multiple-product-validity-page-hint"
        >
          We need to know the time that this product would be valid until
        </span>
        <span
          className="govuk-error-message"
          id="multiple-product-validity-radios-error"
        >
          <span
            className="govuk-visually-hidden"
          />
        </span>
        <FormElementWrapper
          errorClass="govuk-radios--error"
          errorId="multiple-product-validity-error"
          errors={Array []}
        >
          <div
            className="grid-headers-wrapper"
          >
            <div
              className="govuk-heading-s grid-column-header-one-fifth"
            >
              Product Name
            </div>
            <div
              className="govuk-heading-s grid-column-header-one-fifth"
            >
              Product Price
            </div>
            <div
              className="govuk-heading-s grid-column-header-one-fifth"
            >
              Product Duration
            </div>
            <div
              className="govuk-heading-s grid-column-header-one-fifth"
              id="24hr-header"
            >
              24hr
            </div>
            <div
              className="govuk-heading-s grid-column-header-one-fifth header-text-alignment"
              id="calendar-header"
            >
              Calendar
            </div>
          </div>
          <fieldset
            className="govuk-fieldset"
          >
            <div
              className="grid-content-wrapper"
            >
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product0"
              >
                Weekly Ticket
              </label>
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product0"
              >
                £
                50
              </label>
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product0"
              >
                5 days
              </label>
              <div
                className="govuk-radios govuk-radios--inline validity-select-wrapper"
              >
                <div
                  className="govuk-radios__item"
                >
                  <input
                    className="govuk-radios__input"
                    id="twenty-four-hours-row0"
                    name="validity-row0"
                    type="radio"
                    value="24hr"
                  />
                  <label
                    className="govuk-label govuk-radios__label validity-radio-button-margin "
                    htmlFor="twenty-four-hours-row0"
                  >
                    <span
                      className="visually-hidden-label"
                    >
                      24 hour
                    </span>
                  </label>
                </div>
                <div
                  className="govuk-radios__item"
                >
                  <input
                    className="govuk-radios__input"
                    id="calendar-day-row0"
                    name="validity-row0"
                    type="radio"
                    value="endOfCalendarDay"
                  />
                  <label
                    className="govuk-label govuk-radios__label validity-radio-button-margin "
                    htmlFor="calendar-day-row0"
                  >
                    <span
                      className="visually-hidden-label"
                    >
                      Calendar
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </fieldset>
          <fieldset
            className="govuk-fieldset"
          >
            <div
              className="grid-content-wrapper"
            >
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product1"
              >
                Day Ticket
              </label>
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product1"
              >
                £
                2.50
              </label>
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product1"
              >
                1 day
              </label>
              <div
                className="govuk-radios govuk-radios--inline validity-select-wrapper"
              >
                <div
                  className="govuk-radios__item"
                >
                  <input
                    className="govuk-radios__input"
                    id="twenty-four-hours-row1"
                    name="validity-row1"
                    type="radio"
                    value="24hr"
                  />
                  <label
                    className="govuk-label govuk-radios__label validity-radio-button-margin "
                    htmlFor="twenty-four-hours-row1"
                  >
                    <span
                      className="visually-hidden-label"
                    >
                      24 hour
                    </span>
                  </label>
                </div>
                <div
                  className="govuk-radios__item"
                >
                  <input
                    className="govuk-radios__input"
                    id="calendar-day-row1"
                    name="validity-row1"
                    type="radio"
                    value="endOfCalendarDay"
                  />
                  <label
                    className="govuk-label govuk-radios__label validity-radio-button-margin "
                    htmlFor="calendar-day-row1"
                  >
                    <span
                      className="visually-hidden-label"
                    >
                      Calendar
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </fieldset>
        </FormElementWrapper>
      </div>
      <input
        className="govuk-button govuk-button--start"
        id="continue-button"
        type="submit"
        value="Continue"
      />
    </form>
  </main>
</Layout>
`;

exports[`pages multipleProductValidity should render error messaging when errors are passed 1`] = `
<Layout
  description="Multiple Product Validity selection page of the Fares Data Build Tool"
  title="Error: Multiple Product Validity - Fares Data Build Tool"
>
  <main
    className="govuk-main-wrapper app-main-class multiple-product-validity-page"
    id="main-content"
    role="main"
  >
    <form
      action="/api/multipleProductValidity"
      method="post"
    >
      <ErrorSummary
        errors={
          Array [
            Object {
              "errorMessage": "Select one of the two validity options",
              "id": "multiple-product-validity-error",
            },
          ]
        }
      />
      <div
        className="govuk-form-group govuk-form-group--error"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--xl"
        >
          <h1
            className="govuk-fieldset__heading"
            id="multiple-product-validity-page-heading"
          >
            When does the product expire?
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="operator-products-hint"
        >
          Infinity Line
           - 
          2
           products - 
          Adult
        </span>
        <span
          className="govuk-hint"
          id="multiple-product-validity-page-hint"
        >
          We need to know the time that this product would be valid until
        </span>
        <span
          className="govuk-error-message"
          id="multiple-product-validity-radios-error"
        >
          <span
            className=""
          >
            Select one of the two validity options
          </span>
        </span>
        <FormElementWrapper
          errorClass="govuk-radios--error"
          errorId="multiple-product-validity-error"
          errors={
            Array [
              Object {
                "errorMessage": "Select one of the two validity options",
                "id": "multiple-product-validity-error",
              },
            ]
          }
        >
          <div
            className="grid-headers-wrapper"
          >
            <div
              className="govuk-heading-s grid-column-header-one-fifth"
            >
              Product Name
            </div>
            <div
              className="govuk-heading-s grid-column-header-one-fifth"
            >
              Product Price
            </div>
            <div
              className="govuk-heading-s grid-column-header-one-fifth"
            >
              Product Duration
            </div>
            <div
              className="govuk-heading-s grid-column-header-one-fifth"
              id="24hr-header"
            >
              24hr
            </div>
            <div
              className="govuk-heading-s grid-column-header-one-fifth header-text-alignment"
              id="calendar-header"
            >
              Calendar
            </div>
          </div>
          <fieldset
            className="govuk-fieldset"
          >
            <div
              className="grid-content-wrapper"
            >
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product0"
              >
                Weekly Ticket
              </label>
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product0"
              >
                £
                50
              </label>
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product0"
              >
                5 days
              </label>
              <div
                className="govuk-radios govuk-radios--inline validity-select-wrapper"
              >
                <div
                  className="govuk-radios__item"
                >
                  <input
                    className="govuk-radios__input"
                    id="twenty-four-hours-row0"
                    name="validity-row0"
                    type="radio"
                    value="24hr"
                  />
                  <label
                    className="govuk-label govuk-radios__label validity-radio-button-margin validity-radio-button-error-margin"
                    htmlFor="twenty-four-hours-row0"
                  >
                    <span
                      className="visually-hidden-label"
                    >
                      24 hour
                    </span>
                  </label>
                </div>
                <div
                  className="govuk-radios__item"
                >
                  <input
                    className="govuk-radios__input"
                    id="calendar-day-row0"
                    name="validity-row0"
                    type="radio"
                    value="endOfCalendarDay"
                  />
                  <label
                    className="govuk-label govuk-radios__label validity-radio-button-margin validity-radio-button-error-margin"
                    htmlFor="calendar-day-row0"
                  >
                    <span
                      className="visually-hidden-label"
                    >
                      Calendar
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </fieldset>
          <fieldset
            className="govuk-fieldset"
          >
            <div
              className="grid-content-wrapper"
            >
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product1"
              >
                Day Ticket
              </label>
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product1"
              >
                £
                2.50
              </label>
              <label
                className="govuk-label grid-column-content-one-fifth"
                htmlFor="product1"
              >
                1 day
              </label>
              <div
                className="govuk-radios govuk-radios--inline validity-select-wrapper"
              >
                <div
                  className="govuk-radios__item"
                >
                  <input
                    className="govuk-radios__input"
                    id="twenty-four-hours-row1"
                    name="validity-row1"
                    type="radio"
                    value="24hr"
                  />
                  <label
                    className="govuk-label govuk-radios__label validity-radio-button-margin validity-radio-button-error-margin"
                    htmlFor="twenty-four-hours-row1"
                  >
                    <span
                      className="visually-hidden-label"
                    >
                      24 hour
                    </span>
                  </label>
                </div>
                <div
                  className="govuk-radios__item"
                >
                  <input
                    className="govuk-radios__input"
                    id="calendar-day-row1"
                    name="validity-row1"
                    type="radio"
                    value="endOfCalendarDay"
                  />
                  <label
                    className="govuk-label govuk-radios__label validity-radio-button-margin validity-radio-button-error-margin"
                    htmlFor="calendar-day-row1"
                  >
                    <span
                      className="visually-hidden-label"
                    >
                      Calendar
                    </span>
                  </label>
                </div>
              </div>
            </div>
          </fieldset>
        </FormElementWrapper>
      </div>
      <input
        className="govuk-button govuk-button--start"
        id="continue-button"
        type="submit"
        value="Continue"
      />
    </form>
  </main>
</Layout>
`;
